stdlib/messaging/kafka/src/test/java/org/ballerinalang/messaging/kafka/consumer/KafkaConsumerSubscribeToPatternTest.java

```diff
@@ -56,25 +56,26 @@ public void setup() throws IOException {
     }
 
     @Test(description = "Test functionality of getAvailableTopics() function")
-    public void testKafkaConsumerSubscribeToPattern () {
+    public void testKafkaConsumerSubscribeToPattern() {
         result = BCompileUtil.compile(getFilePath(
                 Paths.get(TEST_SRC, TEST_CONSUMER, "kafka_consumer_subscribe_to_pattern.bal")));
-        try {
-            await().atMost(5000, TimeUnit.MILLISECONDS).until(() -> {
-                BValue[] returnBValuesAll = BRunUtil.invoke(result, "funcKafkaGetAvailableTopicsCount");
-                Assert.assertEquals(returnBValuesAll.length, 1);
-                Assert.assertTrue(returnBValuesAll[0] instanceof BInteger);
-                long availableTopicCount = ((BInteger) returnBValuesAll[0]).intValue();
+        await().atMost(10000, TimeUnit.MILLISECONDS).until(() -> {
+            BValue[] returnBValuesAll = BRunUtil.invoke(result, "funcKafkaGetAvailableTopicsCount");
+            Assert.assertEquals(returnBValuesAll.length, 1);
+            Assert.assertTrue(returnBValuesAll[0] instanceof BInteger);
+            long availableTopicCount = ((BInteger) returnBValuesAll[0]).intValue();
 
-                BValue[] returnBValues = BRunUtil.invoke(result, "funcKafkaTestGetSubscribedTopicCount");
-                Assert.assertEquals(returnBValues.length, 1);
-                Assert.assertTrue(returnBValues[0] instanceof BInteger);
-                long topicCount = ((BInteger) returnBValues[0]).intValue();
-                return (topicCount == 0 && availableTopicCount == 0);
-            });
-        } catch (Throwable e) {
-            Assert.fail(e.getMessage());
-        }
+            // Unsubscribe from topics first
```

Comment:
Could you explain the reason for adding the unsubscribing logic here? If the `availableTopicCount` is zero, does it make sense to have already subscribed topics? Shouldn't that also be zero anyway.

Reply:
Yes, valid point. Changed the code accordingly